### PR TITLE
added apisauce utils

### PR DESCRIPTION
**Pull request by damfinkel**
## Summary

Added api config with Apisauce to the utils in React

## Screenshots

image

### CODE REVIEW

File: cookbook-react/src/recipes/utils/apisauce/index.test.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+const onResponseMock = jest.fn();
+
+interface Body {
+  camelCaseField: string;
+}
+interface Response {
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  snake_case_field: string;
+}
```
Comment (pabloferro) [inline cookbook-react/src/recipes/utils/apisauce/index.test.ts:14]: Why do you need this interface?

File: cookbook-react/src/recipes/utils/apisauce/index.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  api.addResponseTransform(response => {
+    if (response.data) {
+      response.data = deserializer.serialize(response.data);
+    }
+  });
+
+  api.addRequestTransform(request => {
+    if (request.data) {
+      request.data = serializer.serialize(request.data);
+    }
```
Comment (pabloferro) [inline cookbook-react/src/recipes/utils/apisauce/index.ts:33]: Is there any reason to not call this outside apiSetup?

This way if someone removes `apiSetup` or doesn't render that component you can loose the transforms.

**Review by pabloferro — COMMENTED** [review]
I have some questions.

File: cookbook-react/src/recipes/utils/apisauce/index.ts (hunk 2 above)
Comment (damfinkel) [reply cookbook-react/src/recipes/utils/apisauce/index.ts:33]: Yep that makes sense, I'll change it

**Review by pabloferro — APPROVED** [review]

**Pull request merged**